Sources/OktaSQLiteStorage/Sources/SQLitePersistentStorage.swift

```diff
@@ -79,21 +79,27 @@ class SQLiteStorage: SQLiteStorageProtocol {
                                 migrator: self.storageMigrator)
                 }
             }
+        } catch let error as SQLiteStorageError {
+            throw error
         } catch {
             throw SQLiteStorageError.sqliteError(error.localizedDescription)
         }
     }
 
     func migrateToTargetVersion<T, S>(fromVersion: T, targetVersionRawValue: Int, migrator: S) throws where T: SchemaVersionType, S: SQLiteMigratable {
+        guard fromVersion.rawValue <= targetVersionRawValue else {
+            throw SQLiteStorageError.migrationError(.downgradeAttempt)
+        }
+
         // swiftlint:disable:next force_cast
         try migrator.willStartIncrementalStorageMigrationSequence(startVersion: fromVersion as! S.Version, endVersion: schema.version as! S.Version)
 
         let allCases = T.allCases.sorted()
         guard let currentVersionIndex = allCases.firstIndex(of: fromVersion) else {
-            throw SQLiteStorageError.generalError("Unexpected db version")
+            throw SQLiteStorageError.migrationError(.badVersion("Unexpected db version"))
         }
         guard let targetVersionIndex = allCases.firstIndex(where: { $0.rawValue == targetVersionRawValue }) else {
-            throw SQLiteStorageError.generalError("Unknown target db version")
+            throw SQLiteStorageError.migrationError(.badVersion("Unknown target db version"))
```

Comment:
So, in these 2 cases, clients will get the same error? 🤔 
Would it be worth differentiating them?